sql/mozfun/newtab/is_default_ui_v1/udf.sql

```diff
@@ -0,0 +1,55 @@
+
+-- newtab.is_default_ui_v1 computes if the newtab opened action is attributed as default_ui
+CREATE OR REPLACE FUNCTION newtab.is_default_ui_v1(
+  event_category STRING,
+  event_name STRING,
+  event_details ARRAY<STRUCT<key STRING, value STRING>>,
+  newtab_homepage_category STRING,
+  newtab_newtab_category STRING
+)
+RETURNS BOOLEAN AS (
+  (
+    event_category = 'newtab'
+    AND event_name = 'opened'
+    AND (
+      (
+        mozfun.map.get_key(event_details, 'source') = 'about:home'
+        AND newtab_homepage_category = 'enabled'
+      )
+      OR (
+        mozfun.map.get_key(event_details, 'source') = 'about:newtab'
+        AND newtab_newtab_category = 'enabled'
+      )
+    )
+  )
```

Comment:
Would be good to update this so it produces `NULL` rather than `FALSE` for events that aren't newtab.opened.  Tried this and it works:
```
    IF(event_category = 'newtab'
    AND event_name = 'opened', 
    (
      (
        mozfun.map.get_key(event_details, 'source') = 'about:home'
        AND newtab_homepage_category = 'enabled'
      )
      OR (
        mozfun.map.get_key(event_details, 'source') = 'about:newtab'
        AND newtab_newtab_category = 'enabled'
      )
    ),
    NULL
  ))
  ```

Reply:
The function is expected to return a BOOLEAN
- `true` when a default ui condition is met
- The function is also un-opinionated on situations that are not "default ui" and would return a `false`
 
This function meets the all the current use cases.

Reply:
Can you explain your reasoning on this? Why is a NULL preferred over a FALSE?

Reply:
I would also like to understand if there are any specific use cases associated with the desired `NULL` outcome, which we may need to consider for a future release.

If there are any such use cases, can those be documented with examples?

Reply:
I think the discussion here is a good example: https://github.com/mozilla/bigquery-etl/pull/7362#discussion_r2093519019

If we ever want to do operations where we want to know when `is_default_ui` is false, the current behavior is counter intuitive.  When the event is not a newtab open, it answer to the question "is this from the default ui?" isn't "No", it's that question doesn't make sense.